source/usage-examples/findOne.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"log"
+)
+
+func main() {
+
+	// Replace the uri string with your MongoDB deployment's connection string.
+	uri := "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority"
+	ctx := context.TODO()
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(uri))
+	if err != nil {
+		panic(err)
+	}
+	// Disconnect the client once the function returns.
+	defer func() {
+		if err = client.Disconnect(ctx); err != nil {
+			panic(err)
+		}
+	}()
+	coll := client.Database("sample_mflix").Collection("movies")
+	var result bson.M
+	opts := options.FindOne()
+	opts.SetProjection(bson.M{"_id": 0, "title": 1, "imdb": 1})
+	opts.SetSort(bson.D{{"imdb.rating", -1}})
+	err = coll.FindOne(ctx, bson.M{"title": "The Room"}, opts).Decode(&result)
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			// This error means your query did not match any documents.
+			return
+		}
+		log.Fatal(err)
+	}
+	fmt.Println(result)
```

Comment:
Nitpick: you could avoid including the fmt package by using log.Println instead.

Reply:
Thank you for the note. I had not considered this, didn't notice that this was the only call to `fmt`, and didn't think about trying another package.

I tested out the `log` package here, and while it does output the correct JSON and avoids an additional import, I am hesitant to use it as it includes a timestamp of when the code was run:

> 2021/04/11 23:33:32 {
    "imdb": {
        "id": 368226,
        "rating": 3.5,
        "votes": 25673
    },
    "title": "The Room"
}